### PR TITLE
Python: Correctly serialize enum with its value (#18327)

### LINKED ISSUE #18327: [BUG][PYTHON] Int Enum serialization is the enum name instead of value
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When using Enum as a parameter, the serialization is the enum name instead of the value. `Enum.NUMBER_0` instead of `0`

##### openapi-generator version

v7.4.0

##### OpenAPI declaration file content or url

```yaml

openapi: 3.1.0
info:
  title: Sample API
  version: 1.0.0
paths:
  /path:
    get:
      parameters:
        - in: query
          name: param
          schema:
            type: array
            items:
              $ref: '#/components/schemas/AnEnum'
      responses:
        200:
          description: Ok
components:
  schemas:
    AnEnum:
      type: integer
      enum:
        - 0
        - 1
```

##### Generation Details

Simple generation in 7.4.0 with -g python

##### Steps to reproduce

This code currently fails

```python
import unittest
from openapi_client import DefaultApi, AnEnum

class TestIssue(unittest.TestCase):

    def test_issue_enum(self):
        assert DefaultApi()._path_get_serialize([AnEnum.NUMBER_0], None,None,None, 0)[1] == '/api/issueX?param=0'
```

##### Suggest a fix
https://github.com/OpenAPITools/openapi-generator/blob/f323a3e788ec932ddc1e31f44f3bd0c8a88f7dbc/modules/openapi-generator/src/main/resources/python/api_client.mustache#L355C25-L355C28 Before testing for primitive types, test for `Enum` and return `obj.value`

In Python, the object is both primitive and enum

--> will push a PR with this fix

### PR DESCRIPTION
**Pull request by BaptisteSaves**
Fixes #18327

Add a serialization case when it is an enum to use its value and not its name.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @cbornet @tomplus @krjakbrjak  @fa0311 @multani

### CODE REVIEW

**Comment by BaptisteSaves** [comment]
Is there any way to add a test that would run the generated Python code ? (IE a pytest)

**Comment by multani** [comment]
@BaptisteSaves you can add a new test, for instance in this folder: https://github.com/OpenAPITools/openapi-generator/tree/master/samples/openapi3/client/petstore/python/tests

**Comment by wing328** [comment]
here is an example on how to add a test: https://github.com/OpenAPITools/openapi-generator/pull/18272/files#diff-865b48c6f08b84b32f8fa35cc3bf6c755b919e5f05c4c2bb157ea9ea589bba0aR622

**Comment by BaptisteSaves** [comment]
@wing328 @multani I added a test ensuring the param is now right.

I have a red build on samples generation but locally `./bin/generate-samples.sh` does not produce any changes. Do you know how to fix it ?

**Comment by wing328** [comment]
please ignore https://github.com/OpenAPITools/openapi-generator/actions/runs/8614615152/job/23608628515?pr=18328 for the time being.

i'll take a look

File: modules/openapi-generator/src/main/resources/python/api_client.mustache [hunk 1]
```diff
@@ -358,6 +358,8 @@ class ApiClient:
         """
         if obj is None:
             return None
+        elif isinstance(obj, Enum):
+            return obj.value
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/python/api_client.mustache:362]: would it be possible to write a unit test for sanitize_for_serialization with enum ?

Comment (BaptisteSaves) [reply modules/openapi-generator/src/main/resources/python/api_client.mustache:362]: Done

**Review by fa0311 — APPROVED** [review]
LGTM! thanks!

**Pull request merged**